core/chains/evm/client/node_lifecycle.go

```diff
@@ -214,33 +227,35 @@ func (n *node) outOfSyncLoop(stuckAtBlockNumber int64) {
 	outOfSyncAt := time.Now()
 
 	lggr := n.lfcLog.Named("OutOfSync")
-	lggr.Debugw("Trying to revive out-of-sync RPC node", "nodeState", n.State())
+	lggr.Debugw("Trying to revive out-of-sync RPC node", "nodeState", n.getState())
 
 	// Need to redial since out-of-sync nodes are automatically disconnected
-	err := n.dial(context.Background())
-	if err != nil {
-		lggr.Errorw("Failed to dial out-of-sync RPC node", "nodeState", n.State())
+	ctx, cancel := utils.ContextFromChan(n.chStop)
+	if err := n.dial(ctx); err != nil {
+		cancel()
```

Comment:
perhaps a `defer cancel()` may be easier?

Reply:
linter is not happy about doing defer in the loops.

Reply:
Fixed